File: app/src/test/java/net/tlalka/android/fiszki/models/types/LanguageTypeTest.java
package net.tlalka.android.fiszki.models.types;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LanguageTypeTest {

    @Test
    public void testCheckLanguageValues() {

        // then
        assertEquals("DB and DTO relations", LanguageType.EN, LanguageType.valueOf("EN"));
        assertEquals("DB and DTO relations", LanguageType.FR, LanguageType.valueOf("FR"));
        assertEquals("DB and DTO relations", LanguageType.DE, LanguageType.valueOf("DE"));
        assertEquals("DB and DTO relations", LanguageType.PL, LanguageType.valueOf("PL"));
    }

    @Test
    public void testCheckLanguageIndexes() {

        // then
        assertEquals("DB index relation", 0, LanguageType.EN.ordinal());
        assertEquals("DB index relation", 1, LanguageType.FR.ordinal());
        assertEquals("DB index relation", 2, LanguageType.DE.ordinal());
        assertEquals("DB index relation", 3, LanguageType.PL.ordinal());
    }
}

File: app/src/test/java/net/tlalka/android/fiszki/model/types/LanguageTypeTest.kt
package net.tlalka.android.fiszki.model.types

import org.junit.Assert.assertEquals
import org.junit.Test

class LanguageTypeTest {

    @Test
    fun testCheckLanguageValues() {

        // then
        assertEquals("DB and DTO relations", LanguageType.EN, LanguageType.valueOf("EN"))
        assertEquals("DB and DTO relations", LanguageType.FR, LanguageType.valueOf("FR"))
        assertEquals("DB and DTO relations", LanguageType.DE, LanguageType.valueOf("DE"))
        assertEquals("DB and DTO relations", LanguageType.PL, LanguageType.valueOf("PL"))
    }

    @Test
    fun testCheckLanguageIndexes() {

        // then
        assertEquals("DB index relation", 0, LanguageType.EN.ordinal.toLong())
        assertEquals("DB index relation", 1, LanguageType.FR.ordinal.toLong())
        assertEquals("DB index relation", 2, LanguageType.DE.ordinal.toLong())
        assertEquals("DB index relation", 3, LanguageType.PL.ordinal.toLong())
    }
}
